client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/test/index.tsx

```diff
@@ -29,319 +29,204 @@ const messages = {
-						blog_url: 'site-url.wordpress.com',
-						notes: 'notes',
-						from_url: siteAddress,
-						...( isBackupRequest
-							? {}
-							: {
-									username,
-									password,
-							  } ),
-					},
-				} );
-			} );
-
-			expectedErrors.forEach( ( text ) => {
-				expect( screen.queryByText( text ) ).toBeInTheDocument();
-			} );
-			notExpectedErrors.forEach( ( text ) => {
-				expect( screen.queryByText( text ) ).not.toBeInTheDocument();
-			} );
-		}
-	);
-
-	it.each( [
-		[
-			'Enter a valid URL.',
-			{
-				code: 'rest_invalid_param',
-				message: 'Test',
-				data: { params: { from_url: 'Invalid Param' } },
-			},
-		],
-		[
-			'Enter a valid username.',
-			{
-				code: 'rest_invalid_param',
-				message: 'Test',
-				data: { params: { username: 'Invalid Param' } },
-			},
-		],
-		[
-			'Invalid input, please check again',
-			{
-				code: 'rest_invalid_param',
-				message: 'Test note',
-				data: { params: { notes: 'Invalid Param' } },
-			},
-		],
-		[
-			'Enter a valid password.',
-			{
-				code: 'rest_invalid_param',
-				message: 'Test',
-				data: { params: { password: 'Invalid Param' } },
-			},
-		],
-		[
-			'An error occurred while saving credentials.',
-			{
-				code: 'rest_invalid_param',
-				message: 'Test',
-				data: { params: { nonexistant: 'Invalid Param' } },
-			},
-		],
-		[
-			'A test message',
-			{
-				code: 'rest_other_error',
-				message: 'A test message',
-				data: {},
+	it( 'creates a credentials using backup file', async () => {
+		const submit = jest.fn();
+		render( { navigation: { submit } } );
+
+		await userEvent.click( backupOption() );
+		await userEvent.type( backupFileInput(), 'backup-file.zip' );
+		await userEvent.click( specialInstructionsButton() );
+		await userEvent.type( specialInstructionsInput(), 'notes' );
+		await userEvent.click( continueButton() );
+
+		//TODO: Ideally we should use nock to mock the request, but it is not working with the current implementation due to wpcomRequest usage that is well captured by nock.
+		expect( wpcomRequest ).toHaveBeenCalledWith( {
+			path: 'sites/site-url.wordpress.com/automated-migration',
+			apiNamespace: 'wpcom/v2/',
+			apiVersion: '2',
+			method: 'POST',
+			body: {
+				blog_url: 'site-url.wordpress.com',
+				migration_type: 'backup',
+				notes: 'notes',
+				from_url: 'backup-file.zip',
 			},
-		],
-		[
-			'An error occurred while saving credentials.',
-			{
-				code: 'rest_other_error_no_message',
-				data: {},
+		} );
+
+		await waitFor( () => {
+			expect( submit ).toHaveBeenCalled();
+		} );
+	} );
+
+	it( 'shows errors on the required fields when the user does not fill the fields', async () => {
+		render();
+		await userEvent.click( continueButton() );
+
+		expect( screen.getByText( messages.urlError ) ).toBeInTheDocument();
+		expect( screen.getByText( messages.usernameError ) ).toBeInTheDocument();
+		expect( screen.getByText( messages.passwordError ) ).toBeInTheDocument();
+	} );
```

Comment:
Do we also need to cover the case that if the user only gives input to the username field but not the password field, we show error only for the password field but not the username one?

Reply:
I don't think so because the react hook form is responsible for refreshing the error object on every submission. 

We would have to do this if we had implemented all of this logic manually.